lib/presentation/journey/todo/bloc/todo_bloc.dart

```diff
@@ -10,8 +10,31 @@ class TodoBloc extends Bloc<TodoEvent, TodoState> {
   TodoBloc({this.todousecase});
 
   @override
-  Stream<TodoState> mapEventToState(TodoEvent event) async* {}
+  Stream<TodoState> mapEventToState(TodoEvent event) async* {
+    switch(event.runtimeType) {
+      case CreateNewTodo:
+        yield* _createTodo(event);
+        break;
+      case Reload:
+        yield* _reload();
+        break;
+      default:
+        break;
+    }
+  }
 
   @override
   TodoState get initialState => TodoInitial();
+
+  Stream<TodoState> _createTodo(CreateNewTodo event) async* {
+    yield CreatingTodo();
+
+    await todousecase.createTodo(event.todo);
+    
+    yield TodoCreated();
```

Comment:
Can we also add the new item to the todos parameter?
so when we navigate back, we will have the item in the list

Reply:
I've added navigate back after succeed creating new data, but I get an issue that the bloc is closed and cannot be added with any event. I am not sure how it can be closed, I don't use bloc.close() in anywhere

Reply:
can you check todo list screen if its closed ?

Reply:
the todo list screen is still in the stack. after succeed create new todo, navigator.pop will bring back to the list screen, but when I add a new event to the bloc to fetch new data, it gets exceptions "Bad state: Cannot add new events after calling close after the bloc is disposed"